proto/cosmos/tx/v1beta1/service.proto

```diff
@@ -14,9 +15,59 @@ service Service {
         option (google.api.http).post = "/cosmos/tx/v1beta1/simulate";
     }
     // GetTx fetches a tx by hash.
-    rpc GetTx(GetTxRequest) returns (GetTxResponse) {
+    rpc GetTx(GetTxRequest) returns (TxResponse) {
         option (google.api.http).get = "/cosmos/tx/v1beta1/tx/{hash}";
     }
+
+    // TxsByEvents fetches txs by event.
+    rpc TxsByEvents(GetTxsEventRequest) returns (TxsByEventsResponse) {
+        option (google.api.http).get = "/cosmos/tx/v1beta1/txs";
+    }
+}
+
+// GetTxsEventRequest is the request type for the Service.TxsByEvents
+// RPC method.
+message GetTxsEventRequest {
+
+    // event is the transaction event type.
+    string event = 1;
+
+    // page is the page number
+    int64 page = 2;
+
+    // limit defines per page limit
+    int64 limit = 3;
```

Comment:
We should ideally use `cosmos.base.query.v1beta1.PageRequest` (and PageResponse in the response).

Then convert `PageRequest` to `page, limit` to pass it into `TxSearch`.